### ISSUE TITLE
Exception on ReadFromEnumerable with NULL values data

### ISSUE THREAD

**Issue by jwood803** [dotnet/machinelearning#2489]
### System information

- **OS version/distro**:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
- **.NET Version (eg., dotnet --info)**: 
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview-009812

### Issue
- **What did you do?**
Loaded the [wine data](https://www.kaggle.com/rajyellow46/wine-quality) into a database and read data to use in `ReadFromEnumerable`

- **What happened?**
Got the below exception when loading the data into `ReadFromEnumerable`. I believe this is because there are null values within the data.
	```
	System.ArgumentOutOfRangeException: 'Could not determine an IDataView type for member 
	Parameter name: rawType'
	```
- **What did you expect?**

### Source code / logs
Have this ML.NET code:
```
var dbData = ReadFromDatabase();

var context = new MLContext();

var mlData = context.Data.ReadFromEnumerable(dbData);
```

And a preview of `dbData` where I think it's failing at:
```
[
  {
    "Type": "white",
    "FixedAcidity": null,
    "VolatileAcidity": 1.0,
    "CitricAcid": 0.0,
    "ResidualSugar": 1.0,
    "Chlorides": 0.0,
    "FreeSulfurDioxide": 29.0,
    "TotalSulfureDioxide": 75.0,
    "Density": 1.0,
    "Ph": 3.0,
    "Sulphates": 0.0,
    "Alcohol": 13.0,
    "Quality": 8.0
  }
]
```

Also, is this the correct way to handle missing data when reading from a database instead of from a CSV file? Or should I be handling it another way?

**Comment by Ivanidzo4ka** [comment]
You are right, it's related to nullable class. This cryptic message comes from here:
https://github.com/dotnet/machinelearning/blob/8444a5a2a4c6b162c35b98cbff95983db3d676c2/src/Microsoft.ML.Data/DataView/InternalSchemaDefinition.cs#L191
where we don't know how to map Nullable<float> to internal type.

@TomFinley  I think we discussed that long time ago, and decided what user should handle that case (`Nullable<Type>`) by themselves. Is it still the case?

**Comment by baruchiro** [comment]
Hi, there is an update about this?

**Comment by dnlhirata** [comment]
Also need a update about this error.

How could I handle?

**Comment by kdcllc** [comment]
@Ivanidzo4ka,
It seems that I am running into the same issue with `float?`

````

**Comment by mikev** [comment]
There is a work-around.  Use the [NoColumn] decorator to exclude that field.  For example, to exclude the nullable Distance field from LoadFromEnumerable:

    public class MyData
    {
        public float SalePrice { get; set; }
        public float SquareFeet { get; set; }
        [NoColumn]
        public int? Distance { get; set; }
    }

**Comment by clydeburgos** [comment]
I had the same issue, and perhaps you're stupid like me, I used decimal to a property and after switching it to float, it fixed my problem. Simple mistakes like these actually returns the same error. So hopefully nobody spends so much time about this because I already did.

From


```
[LoadColumn(1)]
[ColumnName("TotalCommission")]
public decimal TotalCommission { get; set; }
```

to 

```
[LoadColumn(1)]
[ColumnName("TotalCommission")]
public float TotalCommission { get; set; }
```
